### PR TITLE
Save System for Stars. 

### LINKED ISSUE #55: Update save system to handle per night and total "star" scores values across the game
We may want to use some kind of Dict that stores strings as ActNum-ChapterNum and a float value for the result?
^ this one is WIP open to any ideas

Maybe some tool to sum all acts and chapters completed instead of hardcoding total stars? 
Consideration to only save scores to userData if they are better than existing scores
Probably we use this data to actually display user scores in the UI too

### PR DESCRIPTION
**Pull request by SabrinaTheCookie**
closes #55 
inprogress #66 
- Chapters are now saved in a clearer format.
- Total stars earned are saved
- Chapters have star requirements.

### CODE REVIEW

**Review by baconeta — COMMENTED** [review]
I'll have a play with this in editor tonight, but looks good to me so far!

File: Assets/Scripts/GameStructure/Act.cs [hunk 1]
```diff
@@ -40,27 +43,23 @@ private void LoadActData()
...
-        {
-            foreach (ChapterStruct chapter in chapters)
-            {
-                chapter.isComplete = true;
-            }
-        }
-        // We're current at this act
-        else if (userData.currentAct == actNumber)
+        starsEarnedThisAct = 0.0f;
+        foreach (SaveSystem.ChapterSaveData saveData in userData.chapterSaveData)
```
Comment (baconeta) [inline Assets/Scripts/GameStructure/Act.cs:47]: Do you find this readable enough with a LINQ expression?

        // Only compare against chapters in this act
        foreach (var saveData in userData.chapterSaveData.Where(saveData => saveData.actNumber == actNumber))

Comment (baconeta) [reply Assets/Scripts/GameStructure/Act.cs:47]: I ama  tidy kiwi so I prefer them but the if checks are perfectly fine too

**Review by baconeta — APPROVED** [review]
I think I am otherwise happy with basically everything. Clean code!

File: Assets/Scripts/GameStructure/Act.cs (hunk 1 above)
Comment (SabrinaTheCookie) [reply Assets/Scripts/GameStructure/Act.cs]: Didn't know these existed, I like. Prefer not to use var though for readability. 👍 🙇‍♀️

**Pull request merged**